shared/src/main/scala/NumberHelpers.scala

```diff
@@ -103,6 +104,58 @@ object NumberHelpers:
         VList(result.toList*)
       case _ => throw new Exception("Cannot convert to binary")
 
+  def toBase(a: VAny, b: VAny)(using ctx: Context): VAny =
+    (a, b) match
+      case (a: VNum, b: VNum) =>
+        if b.toBigInt == 0 then 0
+        else toBaseDigits(a, b)
+      case (n: VNum, _)  => toBaseAlphabet(n, ListHelpers.makeIterable(b))
+      case (a: VList, _) => VList(a.map(toBase(_, b))*)
+      case _ => throw new Exception("toBase only works on numbers and lists")
+
+  /** Returns value in base len(alphabet) using base 10 [bijective base] */
+  def toBaseAlphabet(value: VNum, alphabet: VIter)(using ctx: Context): VAny =
+    val indexes = toBaseDigits(value, alphabet.iterLength)
+    val alphalist = VList((alphabet match
+      case a: String => a.toString.toList.map(_.toString())
+      case l: VList  => l
+    )*)
+
+    val temp = indexes.map(alphalist.index(_).toString())
+    alphabet match
+      case a: String => temp.mkString("")
+      case l: VList  => l
+
+  def toBaseDigits(value: VNum, base: VNum): VList =
+    if value == VNum(0) then VList(List(VNum(0))*)
+    else if base.toBigInt == -1 then
+      val out = ListBuffer[VNum]()
+      if value.toBigInt > 0 then
+        out += 1
+        for _ <- 1 until value.toInt do
+          out += 0
+          out += 1
+      else
+        val out = ListBuffer[VNum]()
+        for _ <- 1 until value.toInt.abs do
+          out += 1
+          out += 0
+      VList(out.toList*)
+    else
+      val sign = if value.toBigInt < 0 && base.toBigInt > 0 then -1 else 1
+      var current = sign * value.toBigInt
+      if base.toBigInt == 1 then VList(List.fill(current.toInt.abs)(sign)*)
+      else
+        val digits = ListBuffer[VAny]()
+        while current > 0 do
+          var digit = current % base.toBigInt
+          current = current / base.toBigInt
+          if digit < 0 then
+            current += 1
+            digit = digit - base.toBigInt
```

Comment:
```suggestion
            digit -= base.toBigInt
```

Reply:
For some reason, when I tried this, it didn't want to compile